### PR TITLE
Managed Node Group launch template disk size fix

### LINKED ISSUE #1228: Cannot specify Managed Node Group DiskSize when a launch template is created
### What happened?

If you invoke any of the `ManagedNodeGroup` options that result in a custom launch template being created, disk size cannot be set, as AWS expects the disk size to be set on the launch template, not the original request.

### Example

See relevant PR

### Output of `pulumi about`

Latest

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

### PR DESCRIPTION
**Pull request by flostadler**
This fixes the handling of disk size changes when a custom launch template is created for the MNG.
It is based on the Community PR #1229 by @JustASquid, I just added some test fixes on top.

Fixes #1228

### CODE REVIEW

File: examples/examples_nodejs_test.go [hunk 1]
```diff
@@ -227,6 +227,38 @@ func TestAccMNG_withAwsAuth(t *testing.T) {
...
+						// Defined in test managed-ng-disk-size
+						var desiredSizeGB int64 = 50
+						assert.True(t, nodeEphemeralStorage.CmpInt64(desiredSizeGB*1_000_000_000) >= 0)
+					}
+
+				}))
+			},
+		})
+
+	integration.ProgramTest(t, &test)
```
Comment (t0yv0) [inline examples/examples_nodejs_test.go:259]: programTestWithExtraOptions will ignore delete failures

File: nodejs/eks/nodegroup.ts [hunk 2]
```diff
@@ -1742,6 +1742,9 @@ function createManagedNodeGroupInternal(
     let launchTemplate: aws.ec2.LaunchTemplate | undefined;
     if (args.kubeletExtraArgs || args.bootstrapExtraArgs || args.enableIMDSv2) {
         launchTemplate = createMNGCustomLaunchTemplate(name, args, core, parent, provider);
+
+        // Disk size is specified in the launch template.
+        delete nodeGroupArgs.diskSize;
```
Comment (t0yv0) [inline nodejs/eks/nodegroup.ts:1747]: Not quite following control flow here, is diskSize deleted here coming from what the user supplied? Should we warn the user that the supplied value is a no-op in this case?

File: nodejs/eks/nodegroup.ts [hunk 3]
```diff
@@ -1841,9 +1844,19 @@ Content-Type: text/x-shellscript; charset="us-ascii"
         ? { httpTokens: "required", httpPutResponseHopLimit: 2, httpEndpoint: "enabled" }
         : undefined;
 
+    const blockDeviceMappings = args.diskSize ? [
+        {
+            deviceName: "/dev/xvda",
```
Comment (t0yv0) [inline nodejs/eks/nodegroup.ts:1849]: Any comments why "/dev/xvda" could help decipher this in the future.

**Review by t0yv0 — APPROVED** [review]

**Comment by flostadler** [comment]
The test failures are unrelated to this, the CI account is currently clogged up. Working on resolving this right now

**Pull request merged**